### PR TITLE
chore: check task config policies against slots and max_slots

### PR DESCRIPTION
**Pull request by kkunapuli**
## Ticket

[CM-551]

## Description

Need to check both `resources.max_slots` and `resources.slots` when checking if a NTSC workload requests is within constraints.

## Test Plan

Covered by automated tests.

## Checklist

- [ ] Changes have been manually QA'd
- [ ] New features have been approved by the corresponding PM
- [ ] User-facing API changes have the "User-facing API Change" label
- [ ] Release notes have been added as a separate file under `docs/release-notes/`
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses have been included for new code which was copied and/or modified from any external code

[DET-123]: https://hpe-aiatscale.atlassian.net/browse/DET-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[CM-551]: https://hpe-aiatscale.atlassian.net/browse/CM-551?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: master/internal/api_command.go [hunk 1]
```diff
@@ -153,7 +153,7 @@ func (a *apiServer) getCommandLaunchParams(ctx context.Context, req *protoComman
 	// Check submitted config against task config policies.
 	valid, err := configpolicy.CheckNTSCConstraints(ctx, int(cmdSpec.Metadata.WorkspaceID), config, a.m.rm)
 	if !valid {
-		return nil, nil, err
+		return nil, nil, status.Errorf(codes.InvalidArgument, "failed constraint check: %v", err)
```
Comment (kkunapuli) [inline master/internal/api_command.go:156]: @amandavialva01 I had `%w` here at first, but the formatting was super weird. I was reminded of the conversation we had about `%v` vs. `%w` in `fmt.Errorf`. Eventually I'll (hopefully) keep it straight! 😆

Comment (amandavialva01) [reply master/internal/api_command.go:156]: Loll same! 😄 It's definitely a matter of trial and error 😆

File: master/internal/configpolicy/task_config_policy_intg_test.go [hunk 2]
```diff
@@ -111,6 +111,24 @@ func TestValidateNTSCConstraints(t *testing.T) {
...
+		_, err := db.Bun().NewInsert().Model(&w).Exec(context.Background())
+		require.NoError(t, err)
+		addConstraints(t, user, &w.ID, *constraints)
+
+		resourceManager := mocks.ResourceManager{}
+		resourceManager.On("SmallerValueIsHigherPriority", mock.Anything).Return(true, nil)
+
+		config := defaultConfig()
+		config.Resources.Slots = *config.Resources.MaxSlots
+		config.Resources.MaxSlots = nil // ensure only slots is set
```
Comment (amandavialva01) [inline master/internal/configpolicy/task_config_policy_intg_test.go:126]: I like this setup! Makes the test function explicit

**Review by amandavialva01 — APPROVED** [review]
LGTM! Awesome work

**Pull request merged**